casper/src/main/scala/coop/rchain/casper/merging/ConflictSetMerger.scala

```diff
@@ -44,6 +50,54 @@ object ConflictSetMerger {
         .getOrElse(Set.empty)
     }
 
+    def calMergedResult(
+        branch: Branch,
+        originResult: Map[Blake2b256Hash, Long]
+    ): Map[Blake2b256Hash, Long] = {
+      val diff = branch.map(mergeableChannels).toList.combineAll
+      diff.foldLeft(originResult) {
+        case (ba, br) =>
+          val result = Math.addExact(ba.getOrElse(br._1, 0L), br._2)
+          assert(result >= 0, "merged result negative")
```

Comment:
Better to catch error inside this function and return result in Option.

Reply:
I made some modifications. Please take a look